File: 백준/Silver/11399. ATM/ATM.py
import sys

n = sys.stdin.readline()

answer = 0
time_prv = 0

times = list(map(int, sys.stdin.readline().split()))

times.sort()

for i in range(len(times)):
    if i == 0:
        answer += times[i]
        time_prv = times[i]
        continue
    answer += times[i] + time_prv
    time_prv += times[i]

print(answer)

File: 백준/Silver/11399. ATM/ATM.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] array = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i=0; i<n; i++){
            array[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(array);

        int sum = 0;
        int answer = 0;
        for (int i=0; i<n; i++){
            sum += array[i];
            answer += sum;
        }
        System.out.println(answer);
    }
}
